ffi/src/lib.rs

```diff
@@ -237,6 +240,23 @@ where
     type OutPtr = *mut Self;
 }
 
+/// Wrapper around struct/enum opaque pointer. When wrapped with the [`ffi`] macro in the
+/// crate linking dynamically to some`cdylib`, it replaces struct/enum body definition
+#[derive(Debug, Clone, Copy)]
+#[repr(transparent)]
+pub struct Opaque {
+    __data: [u8; 0],
```

Comment:
What is the purpose of this field?

Reply:
this is rust idiom to represent `extern` type

Reply:
Ok, it's ZST, compitable with `#[repr(C)]` and we can't construct `Opaque` because of private fields.